servicemanager/servicemanager.go

```diff
@@ -474,6 +481,44 @@ func (sm *ServiceManager) doRemoveService(serviceID string) error {
 	return nil
 }
 
+func (sm *ServiceManager) validateTTLs() {
+	ttlRemoveServices := time.NewTicker(TTLRemoveServices)
+	defer ttlRemoveServices.Stop()
+
+	for {
+		select {
+		case <-ttlRemoveServices.C:
+			if err := sm.removeCachedServices(); err != nil {
+				log.Errorf("Can't remove cached services: %v", err)
+			}
+
+		case <-sm.ttlStopChannel:
+			return
+		}
+	}
+}
+
+func (sm *ServiceManager) removeCachedServices() error {
+	services, err := sm.serviceInfoProvider.GetAllServices()
```

Comment:
I would implement `GetCachedServices` storage API and use it here.

Reply:
I wrote this function based on previous comments, I think that it would be better to have a function that returns all versions of services, this will be useful in the future for other use cases. `GetCachedServices` more specific function